src/backend/distributed/planner/query_pushdown_planning_utils.c

```diff
@@ -0,0 +1,1704 @@
+/*-------------------------------------------------------------------------
+ *
+ * query_pushdown_planning_utils.c
+ *
+ * Routines for creating pushdown plans for queries.
+ *
+ * Copyright (c) 2012-2016, Citus Data, Inc.
```

Comment:
Hmm, I think we might use `Copyright (c) 2018, Citus Data, Inc.`. Well, I'm not sure about that 😄 Could you find someone who knows this and maybe reply here what should we do?